src/Base/UnitPyImp.cpp

```diff
@@ -206,21 +207,18 @@ PyObject* UnitPy::richCompare(PyObject* v, PyObject* w, int op)
 
 Py::String UnitPy::getType() const
 {
-    return {getUnitPtr()->getTypeString().toUtf8(), "utf-8"};
+    return {getUnitPtr()->getTypeString(), "utf-8"};
 }
 
 Py::Tuple UnitPy::getSignature() const
 {
-    const UnitSignature& Sig = getUnitPtr()->getSignature();
     Py::Tuple tuple(8);
-    tuple.setItem(0, Py::Long(Sig.Length));
-    tuple.setItem(1, Py::Long(Sig.Mass));
-    tuple.setItem(2, Py::Long(Sig.Time));
-    tuple.setItem(3, Py::Long(Sig.ElectricCurrent));
-    tuple.setItem(4, Py::Long(Sig.ThermodynamicTemperature));
-    tuple.setItem(5, Py::Long(Sig.AmountOfSubstance));
-    tuple.setItem(6, Py::Long(Sig.LuminousIntensity));
-    tuple.setItem(7, Py::Long(Sig.Angle));
+    Unit* self = getUnitPtr();
+
+    for (auto i = 0; i < 7; i++) {
```

Comment:
This is wrong! The last iteration is done for i=6 and it stops with i=7. But the old code also added a value to the last item of the tuple.

The 7 should be replaced with tuple.size()

Reply:
Stupid mistake, indeed. Fixed.